### PR TITLE
Always send SIGTERM to child + handle SIGQUIT

### PR DESCRIPTION
**Pull request by pierresouchay**

### CODE REVIEW

File: runner/signals_darwin.go [hunk 1]
```diff
@@ -9,9 +10,10 @@ import (
 
 func signalWatch(cmd *exec.Cmd) {
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
 	go func() {
 		sig := <-signals
-		syscall.Kill(cmd.Process.Pid, sig.(syscall.Signal))
+		fmt.Println("Received Signal", sig, "sending SIGTERM to child", cmd.Process.Pid)
+		syscall.Kill(cmd.Process.Pid, syscall.SIGTERM)
```
Comment (ShimmerGlass) [inline runner/signals_darwin.go:17]: You should propagate the same signal to avoid surprises

File: runner/signals_linux.go [hunk 2]
```diff
@@ -9,9 +10,10 @@ import (
 
 func signalWatch(cmd *exec.Cmd) {
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
 	go func() {
 		sig := <-signals
-		syscall.Kill(cmd.Process.Pid, sig.(syscall.Signal))
+		fmt.Println("Received Signal", sig, "sending SIGTERM to child", cmd.Process.Pid)
+		syscall.Kill(cmd.Process.Pid, syscall.SIGTERM)
```
Comment (ShimmerGlass) [inline runner/signals_linux.go:17]: Same

**Review by ShimmerGlass — CHANGES_REQUESTED** [review]

File: runner/signals_linux.go (hunk 2 above)
Comment (pierresouchay) [reply runner/signals_linux.go:17]: Well, in that case, it deserves a refactoring: it SIGQUIT is sent, then SIGTERM, it does not behaves as it should If I am not mistaken

Comment (ShimmerGlass) [reply runner/signals_linux.go:17]: Yep, signals should be handled in a for loop. You can just propagate the received signal each time.

**Pull request closed without merging**